### ISSUE TITLE
[BUG] dealing with `sklearn 1.2` deprecation warnings

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2143]
A number of `sklearn` calls raise deprecation warnings and will turn into bugs when `sklearn` releases version 1.2.
These should be fixed.

- [x] replacing `float` variable names by `str` in `pandas.DataFrame` passed:

```
classification/feature_based/_summary_classifier.py: 2 warnings
classification/feature_based/tests/test_summary_classifier.py: 2 warnings
classification/tests/test_all_classifiers.py: 6 warnings
tests/test_all_estimators.py: 23 warnings
  C:\Miniconda\envs\test\lib\site-packages\sklearn\utils\validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['float', 'str']. An error will be raised in 1.2.
```

- [x] replacing `normalize` by use of `StandardScaler`:

```
transformations/panel/rocket/tests/test_MiniRocket.py::test_minirocket_on_gunpoint
transformations/panel/rocket/tests/test_MiniRocketMultivariate.py::test_minirocket_multivariate_on_basic_motions
transformations/panel/rocket/tests/test_MultiRocket.py::test_multirocket_on_gunpoint
transformations/panel/rocket/tests/test_MultiRocketMultivariate.py::test_multirocket_multivariate_on_basic_motions
transformations/panel/rocket/tests/test_Rocket.py::test_rocket_on_gunpoint
  C:\Miniconda\envs\test\lib\site-packages\sklearn\linear_model\_base.py:145: FutureWarning: 'normalize' was deprecated in version 1.0 and will be removed in 1.2.
  If you wish to scale the data, use Pipeline with a StandardScaler in a preprocessing stage. To reproduce the previous behavior:
  
  from sklearn.pipeline import make_pipeline
  
  model = make_pipeline(StandardScaler(with_mean=False), _RidgeGCV())
```

**Comment by KishenSharma6** [comment]
Hi @fkiraly , I'd like to work on this bug.

**Comment by fkiraly** [comment]
alright, @kishensharma6, thanks! Pick one of the two to start with, I suggest.

**Comment by KishenSharma6** [comment]
Thanks @fkiraly ,  the pr #2167 is for replacing normalize by use of StandardScaler. 

Please let me know if that pr needs anything additional, I will begin on replacing float variable names by str in pandas.DataFrame passed:

**Comment by hmtbgc** [comment]
@KishenSharma6 I am sorry that I have PR the first part. You have done a good job. Thanks for your contribution!

**Comment by KishenSharma6** [comment]
@hmtbgc no worries, thank you so much! I'm having issues building sktime from source so your assistance was much needed and greatly appreciated :)

**Comment by jaivanti** [comment]
Hi! I would like to work on this?

**Comment by jaivanti** [comment]
If possible can you please brief more about the the 1st part....Like as a beginner just want to clarify more on it. Thanks

**Comment by KishenSharma6** [comment]
@fkiraly We can mark this one as completed. PR #2167 and #2190 for reference

@jaivanti sorry for the late reply, we were able to complete this one. Feel free to work on another issue. You can reference #2190 to see how @hmtbgc solved the issue (namely changing input data types).

**Comment by fkiraly** [comment]
Yes, indeed, @KishenSharma6, this is done. Forgot to update and close  after #2190, thanks for pointing this out!
